### PR TITLE
Inject PluginRegistry to Puppet

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes a part of #2579 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] ~Added relevant unit tests?~
* [x] Have you successfully tested your changes locally? Elaborate:
    > Test by running the Island and Agent and making sure nothing crashes.
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

**Review by ilija-lazoroski — APPROVED** [review]
If we don't need to use `DIContainer`, LGTM!

File: monkey/infection_monkey/monkey.py [hunk 1]
```diff
@@ -295,7 +296,8 @@ def _build_server_list(self, relay_port: int) -> Sequence[str]:
         return list(ordered_servers.keys())
 
     def _build_puppet(self) -> IPuppet:
-        puppet = Puppet(self._agent_event_queue)
+        plugin_registry = PluginRegistry()
+        puppet = Puppet(self._agent_event_queue, plugin_registry)
 
         puppet.load_plugin(
```
Comment (VakarisZ) [inline monkey/infection_monkey/monkey.py:302]: I don't understand why it's the responsibility of the puppet to load the plugin? Puppet doesn't store plugins, so it shouldn't have `load_plugin` method 🤔 What we should do is load the plugins into the registry and then pass the registry to the puppet

Comment (mssalvatore) [reply monkey/infection_monkey/monkey.py:302]: It'll go away eventually, but just leave it for now. It's not worth changing.

File: monkey/infection_monkey/puppet/puppet.py [hunk 2]
```diff
@@ -14,18 +14,19 @@
...
 EMPTY_FINGERPRINT = FingerprintData(None, None, [])
 
 logger = logging.getLogger()
 
 
 class Puppet(IPuppet):
-    def __init__(self, agent_event_queue: IAgentEventQueue) -> None:
-        self._plugin_registry = PluginRegistry()
+    def __init__(
+        self, agent_event_queue: IAgentEventQueue, plugin_registry: PluginRegistry
```
[the remark this answers is not in the document]
Comment (mssalvatore) [reply monkey/infection_monkey/puppet/puppet.py:27]: Ideally, yes. Practically speaking, 🤷. I'm fine either way.

**Review by mssalvatore — APPROVED** [review]

**Comment by mssalvatore** [comment]
@VakarisZ 

> I think we need to define the plugin repository as well. We should then build the repository, with it we can build the registry and pass it to the puppet.

We don't need a repository for this. The registry can handle it. It's just managing plugins in memory at runtime. There's no persistence and nothing else needs access to the plugins.

> PluginRepository is responsible for retrieving the plugins and translating them into python objects that the Registry can use.

This is effectively a matter of deserialization. The `IIslandHTTPClient` handles deserialization for all other objects; plugins shouldn't be any different.

**Comment by mssalvatore** [comment]
@ilija-lazoroski 

> If we don't need to use DIContainer

We can get by without it. Adding it is a project for next release.

**Review by VakarisZ — APPROVED** [review]

**Pull request merged**